StandardLibraryEssentials/StepForwardWithSTL/LeaderboardCreation/src/leaderboard.cpp

```diff
@@ -0,0 +1,115 @@
+#include "leaderboard.hpp"
+
+void Leaderboard::retrieveScores() {
+    std::ifstream file(filepath);
+    if (file.is_open()) {
+        std::string line;
+        while (getline(file, line)) {
+            std::stringstream ss(line);
+            std::string name;
+            unsigned int score;
+            getline(ss, name, ',');
+            ss >> score;
+            this->scores.insert({score, name});
+        }
+        file.close();
+    } else {
+        std::cout << "Unable to open file: " << this->filepath << std::endl;
+    }
+}
+
+void Leaderboard::addScore(unsigned int score, std::string name) {
+    scores.insert({score, name});
+    if (scores.size() == 11) {
+        scores.erase(std::prev(scores.end()));
+    }
+}
+
+void Leaderboard::saveScores() {
+    std::ofstream
+            file(filepath);
+    if (file.is_open()) {
+        for (const auto &[score, name]: scores) {
+            file << name << "," << score << std::endl;
+        }
+        file.close();
+    } else {
+        std::cout << "Unable to open file: " << this->filepath << std::endl;
+    }
+}
+
+void Leaderboard::updateScore(unsigned int score) {
+    retrieveScores();
+    addScore(score);
+    saveScores();
+}
+
+sf::Text createText(const std::string &str, const sf::Font &font, const unsigned int size, const float x, const float y) {
+    sf::Text text;
+    text.setFont(font);
+    text.setCharacterSize(size);
+    text.setFillColor(sf::Color::White);
+    text.setString(str);
+    text.setPosition(x, y);
+    return text;
+}
+
+void Leaderboard::drawLeaderboard(sf::RenderWindow &window, const TextureManager &textureManager, ScoreMap scores) {
+    // show leaderboard frame
+    sf::Sprite frame;
+    frame.setTexture(*textureManager.getTexture(GameTextureID::LEADERBOARD));
+    frame.setPosition(100, 75);
+    window.draw(frame);
+
+    sf::Font font;
+    font.loadFromFile("resources/pixelLetters.ttf");
+
+    sf::Text header = createText("LEADERBOARD", font, 40, 300, 80);
```

Comment:
Please do not use magic constants. 
Extract all the constants and make them named, e.g. 
```
const int TITLE_FONT_SIZE = 40;
```